rdmd.d

```diff
@@ -142,14 +139,16 @@ int main(string[] args)
     string[] eval;     // set by --eval
     bool makeDepend;
     string makeDepFile;
+    string requestedCompiler;
```

Comment:
Do you think it's possible to linearize this computation a bit and reduce the amount of needed state?

E.g.

```d
string compiler;
getopt(..., &compiler);
if (!compiler)
    compiler = environment.get("DC", null);
if (!compiler)
{
    compiler = defaultCompiler;
    // do the lookup in the local directory here, because it only applies here
}
```

Reply:
Pretty much all of that logic can be moved back into main if we want to do that.

My main concern in doing this is that it is already sitting at around ~270LOC and doing an awful lot there.

Reply:
As long as it's all in one place, doesn't need more state than necessary, and the priorities of the rules used to make the decision all go in only one direction (e.g. highest-priority first).